django/thunderstore/repository/utils.py

```diff
@@ -56,3 +63,51 @@ def unpack_serializer_errors(field, errors, error_dict=None):
     else:
         error_dict[field] = str(errors)
     return error_dict
+
+
+@cache_function_result(cache_until=CacheBustCondition.any_package_updated)
+def get_listing(
+    owner: Union[UploaderIdentity, str, None],
+    name: Union[str, None],
+    community: Union[Community, None],
+) -> PackageListing:
+    if isinstance(owner, str):
+        owner = get_object_or_404(UploaderIdentity, name=owner)
+    filters = Q()
+    if owner:
+        filters.add(Q(package__owner=owner), Q.AND)
+    if name:
+        filters.add(Q(package__name=name), Q.AND)
+    if community:
+        filters.add(Q(community=community), Q.AND)
+    package_listing = (
+        PackageListing.objects.active()
+        .filter(filters)
+        .select_related(
+            "package",
+            "package__owner",
+            "package__latest",
+        )
+        .prefetch_related(
+            "categories",
+        )
+        .first()
+    )
+    # Try finding listing in another community
+    if not package_listing:
+        package_listing = (
+            PackageListing.objects.active()
+            .filter(Q(Q(package__owner=owner), Q(package__name=name)))
```

Comment:
Moreover I think this'll be cleaner to do at the call site rather than here, so how about splitting this logic into two functions? One which just gets & caches whatever matching its parameters (like the old implementation), and another which calls that & re-tries without community if not found?

Reply:
Yeah true, that would be cleaner. Tunnel vision hit hard this implementation.

Reply:
Also as for the disparity, I have no idea :D Probably my subconscious didn't register, that I'd have to do those conditional checks twice.